c/tskit/trees.h

```diff
@@ -265,15 +265,188 @@ typedef struct {
 @defgroup TREESEQ_API_GROUP Tree sequence API
 @{
 */
+
+/**
+@brief Initialises the tree sequence by (optionally) copying the supplied
+table collection, checking its integrity then building index maps.
+
+@rst
+This must be called before any operations are performed on the tree sequence.
+See the :ref:`sec_c_api_overview_structure` for details on how objects
+are initialised and freed.
+
+**Options**
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_TAKE_TABLES
+    By default the table collection is copied, however if this flag is
+    specified the table collection will be taken and owned with a reference
+    held by this tree sequence. The table collection will be free'd with
+    this tree sequence.
+
+TSK_BUILD_INDEXES
+    If specified edge indexes will be built and stored in the table collection
+@endrst
+
+@param self A pointer to an uninitialised tsk_table_collection_t object.
+@param tables A pointer to a tsk_table_collection_t
+@param options Allocation time options. See above for details.
+@return Return 0 on success or a negative value on failure.
+*/
 int tsk_treeseq_init(
     tsk_treeseq_t *self, tsk_table_collection_t *tables, tsk_flags_t options);
 
+/**
+@brief Load a tree sequence from a file path.
+
+@rst
+Loads the data from the specified file into this tree sequence.
+The tree sequence is also initialised.
+The resources allocated must be freed using
+:c:func:`tsk_treeseq_free` even in error conditions.
+
+If the file contains multiple tree sequences, this function will load
+the first. Please see the :c:func:`tsk_treeseq_loadf` for details
+on how to sequentially load tree sequences from a stream.
+
+If the TSK_LOAD_SKIP_REFERENCE_SEQUENCE option is set, the table collection is
+read without loading the reference sequence.
+
+**Options**
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_LOAD_SKIP_REFERENCE_SEQUENCE
+    Do not load reference sequence.
+
```

Comment:
Is SKIP_TABLES an option here? What about TSK_NO_CHECK_INTEGRITY?

Reply:
Neither are a valid option. We could think about adding `TSK_NO_CHECK_INTEGRITY` but `SKIP_TABLES` makes no sense I think.

Reply:
I think SKIP_TABLES will be accepted and work, you'll just get a tree_sequence that only has top-level stuff. This should be documented I think, as it was part of the original design that you could quickly get access to top-level stuff from either the ts or the tables interface.

Just checked, and it's tested in ``test_file_format.c``

Reply:
Maybe we could dedupe this a bit by referring to ``tsk_table_collection_load`` and say that this is basically the same, unless ``TSK_BUILD_INDEXES`` is specfied (which will additionally build the indexes)?

Same for loadf below.

Reply:
Great, didn't realise/remember that SKIP_TABLES worked in that case, having a brain foggy day.
I've added a link to the table methods.